x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/notifications/AbstractAuditor.java

```diff
@@ -83,4 +168,21 @@ private XContentBuilder toXContentBuilder(ToXContent toXContent) {
             throw new RuntimeException(e);
         }
     }
+
+    private void writeBacklog() {
+        BulkRequest bulkRequest = new BulkRequest();
+        ToXContent doc = backlog.poll();
+        while (doc != null) {
+            bulkRequest.add(indexRequest(doc));
+            doc = backlog.poll();
+        }
```

Comment:
I know there will be complexity, but we should make sure we don't breach some limit (10k, etc.) The size of the messages will probably be small enough so we don't have to worry about the bulk index byte size.
But the queue length is unbounded. We should check bounds somewhere.

Reply:
I do think adding a sane limit (like 100 or something) would be simple enough. The `!= null` while loop could still be on the outside, just need to have an inner conditional (and conditional afterwards) that calls `client.bulk` and resets the request. 

Will protect against a weird, and pretty nasty edge case.

Reply:
I pushed a change to limit the buffer size to 1000, after 1000 messages the oldest ones start getting dropped from the queue. This simplifies the code as there will be at most 1000 messages in the bulk index request.

Reply:
@davidkyle sorry! I didn't see that :D